### PR TITLE
Save secrets

### PR DESCRIPTION
**Pull request by darrenmcc**

### CODE REVIEW

File: gcpvault.go [hunk 1]
```diff
@@ -77,39 +76,86 @@ func GetSecrets(ctx context.Context, cfg Config) (map[string]interface{}, error)
...
-	vClient, err := api.NewClient(vcfg)
+	// fetch secrets
+	secrets, err := vClient.Logical().Read(cfg.SecretPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get secrets")
+	}
+	return secrets.Data, nil
+}
+
+func PutSecrets(ctx context.Context, cfg Config, secrets map[string]interface{}) error {
```
Comment (jprobinson) [inline gcpvault.go]: Mind putting some docs on these new functions?

File: gcpvault.go [hunk 2]
```diff
@@ -77,39 +76,86 @@ func GetSecrets(ctx context.Context, cfg Config) (map[string]interface{}, error)
...
+func GetVersionedSecrets(ctx context.Context, cfg Config) (map[string]interface{}, error) {
+	secs, err := GetSecrets(ctx, cfg)
+	if err != nil {
+		return nil, err
+	}
+	// versioned secrets are contained under a 'data' key
+	return secs["data"].(map[string]interface{}), nil
+}
+
+func PutVersionedSecrets(ctx context.Context, cfg Config, secrets map[string]string) error {
```
Comment (jprobinson) [inline gcpvault.go]: should this be `map[string]interface{}` to be symmetric with the other funcs?

**Comment by darrenmcc** [comment]
Ready for another look

File: gcpvault.go [hunk 3]
```diff
@@ -77,39 +77,88 @@ func GetSecrets(ctx context.Context, cfg Config) (map[string]interface{}, error)
...
+	if err != nil {
+		return errors.Wrap(err, "unable to login to vault")
+	}
+
+	_, err = vClient.Logical().Write(cfg.SecretPath, secrets)
+	return errors.Wrap(err, "unable to make vault request")
+}
+
+// PutVersionedSecrets writes versioned secrets to Vault at the configured path.
+// The path should be in the form `org/repo/versioned/data/my-secret`
```
Comment (jprobinson) [inline gcpvault.go]: From looking at the Vault docs, this path convention is NYT specific.

Can we add a new config option for the "versioned" substring to look for, but have it default to `/versioned/` via the envconfig tag?

**Pull request merged**